modules/ROOT/pages/dataweave-type-selector.adoc

```diff
@@ -0,0 +1,206 @@
+= Type Selector
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Using the dot selector (`.`) over DataWeave xref:dataweave-type-system.adoc#composite-types[types] enables you to declare new types from existing ones. Combined with the xref:dataweave-selectors-reusing-types.adoc[module loader], DataWeave can also load and translate declarations from a custom module file into DataWeave type directives that can be accessed in the same way as types from any other DataWeave module.
+
+== Example: Declaring a New Type from an Existing Type
+The following example declares the `NameType` from the `User` type. Using the dot selector in `User.name` to declare the `name` variable a `NameType` enables the `typeOf()`  function to return the basic type, `"String"`. Notice how the DataWeave variable `name` is declared as a `NameType` value "Seba".
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+type User = {name: String}
+type NameType = User.name
+var name: NameType = "Seba"
+---
+typeOf(name)
+----
+
+.Output: 
+----
+"String"
+----
+
+== Use the Dot Selector (`.`) Over DataWeave Types
```

Comment:
```suggestion
== Use the Dot Selector (`.`) Over DataWeave Types
```
Not sure that this heading is needed here. The H3 subsections could become H2 headings that follow the same heading pattern (e.g., "Doing X"). Suggestions below.

Reply:
If the subsections here are H2 level headings, they will appear as links on the right side of the page, which could enhance the usability of this page. Otherwise people will have to scroll down the page to see that we can do selection with namespaces, etc. I'm thinking that people can scan and see what's on the page with less scrolling. (which could be another argument for breaking up that selector page into separate pages)